### PR TITLE
[FrameworkBundle] Added option to specify the event dispatcher in debug:event-dispatcher

### PR DESCRIPTION
**Pull request by TimoBakx**
| Q             | A
| ------------- | ---
| Branch?       | 5.x
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | N/A
| License       | MIT
| Doc PR        | symfony/symfony-docs#14650
| Tags | #SymfonyHackday

### Description:
In the recent security improvements, new event dispatchers were added (one for each firewall). These were not visible in the `debug:event-dispatcher` command. This PR adds an option to define a event dispatcher to show the events (and listeners) for that specific event dispatcher. Without any options, the default dispatcher is shown (identical to previous working)

In order to be able to fetch specific event dispatchers, I tagged all event dispatchers with a new `kernel.event-dispatcher` tag.

### Usage:
`bin/console debug:event-dispatcher --dispatcher=[servicename] [other options] [event]`

### Subtasks:
- [x] Add `dispatcher` option
- [x] Add `kernel.event_dispatcher` tag to all event dispatchers
- [x] Add autoconfigure tag for services implementing `EventDispatcherInterface`
- [x] Update Descriptor classes to specify the chosen dispatcher
- [x] Update changelog
- [x] Add documentation

### CODE REVIEW

**Review by wouterj — COMMENTED** [review]
Thank you @TimoBakx! We completely forgot to update the debug and web profiler features for multiple event dispatchers.

I think this is a great feature. We can improve the DX later on by adding a `debug:firewall` command or the like (allowing users to specify the firewall name, rather than a weird `security.event_dispatcher.<firewall name>` string - #39321).

The only thing I'm a bit worried about is that the `kernel.event_dispatcher` tag is going to disrupt tag autocompletion for `kernel.event_subscriber`/`kernel.event_listener` tags (causing people to use the wrong tag). Can we maybe find a name that doesn't look similar to another tag?

**Comment by TimoBakx** [comment]
Hi @wouterj, that is a good point.

@weaverryan and I picked `kernel.event_dispatcher` for the consistency with other tags. My other ideas were `event.dispatcher` or `event_dispatcher`, but those seem a bit short and out of place.

`kernel.dispatcher` won't do due to the message bus also dispatching.

I'm a bit out of ideas on this one.

**Comment by derrabus** [comment]
Would it make sense to enable auto-configuration for the new tag?

**Comment by TimoBakx** [comment]
> Would it make sense to enable auto-configuration for the new tag?

It is. See https://github.com/symfony/symfony/pull/39276/files#diff-e4f52d5033296421d02e360c8355aa1931bdfb0c12579f4815ea02b724322c0cR483

**Comment by derrabus** [comment]
Oh right, sorry. 🙈

**Comment by wouterj** [comment]
> I'm a bit out of ideas on this one.

I'm also a bit of out of ideas. Maybe `event_dispatcher.dispatcher`? (to match the `<component name>.<subject>` convention)

**Comment by TimoBakx** [comment]
> Maybe `event_dispatcher.dispatcher`? (to match the `<component name>.<subject>` convention)

That sound good to me. I'll make the change.

File: src/Symfony/Bundle/FrameworkBundle/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
 CHANGELOG
 =========
 
+5.3.0
+-----
+
+ * Added `dispatcher` option to `debug:event-dispatcher`
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/CHANGELOG.md]: Added the `dispatcher`[...]

File: src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/MarkdownDescriptor.php [hunk 2]
```diff
@@ -287,8 +287,14 @@ protected function describeContainerEnvVars(array $envs, array $options = [])
     protected function describeEventDispatcherListeners(EventDispatcherInterface $eventDispatcher, array $options = [])
     {
         $event = \array_key_exists('event', $options) ? $options['event'] : null;
+        $dispatcherServiceName = \array_key_exists('dispatcher_service_name', $options) ? $options['dispatcher_service_name'] : null;
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/MarkdownDescriptor.php]:
```suggestion
        $dispatcherServiceName = $options['dispatcher_service_name'] ?? null;
```

Comment (TimoBakx) [reply src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/MarkdownDescriptor.php:290]: While I'm at it, should I do the same with the line above it? So they both look the same?

File: src/Symfony/Bundle/FrameworkBundle/Command/EventDispatcherDebugCommand.php [hunk 3]
```diff
@@ -74,21 +77,35 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $io = new SymfonyStyle($input, $output);
 
         $options = [];
+        $dispatcherServiceName = $input->getOption('dispatcher');
+        if (!$this->dispatchers->has($dispatcherServiceName)) {
+            $io->getErrorStyle()->warning(sprintf('Event dispatcher "%s" is not available.', $dispatcherServiceName));
+
+            return 0;
```
Comment (wouterj) [inline src/Symfony/Bundle/FrameworkBundle/Command/EventDispatcherDebugCommand.php]: shouldn't this be `return 1;` to indicate an error?

**Review by wouterj — APPROVED** [review]

**Review by derrabus — APPROVED** [review]

**Comment by derrabus** [comment]
Thank you @TimoBakx.

**Pull request merged**